> > ### Safer C++ Build [#26369](https://ews-build.webkit.org/#/builders/120/builds/26369) ([7c6d50b](https://github.com/WebKit/WebKit/commit/7c6d50b7f088bc308aced762f4de72dbb396f1db))
> > ❌ Found [0 new failure](https://ews-build.s3-us-west-2.amazonaws.com/macOS-Safer-CPP-Checks-EWS/7c6d50b7-26369/scan-build-output/new-results.html). Please address these issues before landing. See [WebKit Guidelines for Safer C++ Programming](https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines). (cc @rniwa)
> 
> @briannafan FYI

Reading the logs, the issue wasn't fixed:

/Volumes/Data/worker/macOS-Safer-CPP-Checks-EWS/build/Source/WebCore/css/CSSFontFace.cpp:253:61: warning: Implicitly captured raw-pointer 'rangeValue' to ref-counted type or CheckedPtr-capable type is unsafe [webkit.UncountedLambdaCapturesChecker]
  253 |         return Style::fontStyleAngleFromCSSValueDeprecated(*rangeValue->obliqueValues->itemWithoutBoundsCheck(index));
      |                                                             ^

--- a/Source/WebCore/SaferCPPExpectations/UncountedLambdaCapturesCheckerExpectations
+++ b/Source/WebCore/SaferCPPExpectations/UncountedLambdaCapturesCheckerExpectations
@@ -58,7 +58,6 @@ animation/KeyframeEffect.cpp
 bindings/js/JSDOMPromiseDeferred.cpp
 contentextensions/ContentExtensionsBackend.cpp
 crypto/SubtleCrypto.cpp
-css/CSSFontFace.cpp
 css/CSSPendingSubstitutionValue.cpp
 css/CSSValue.cpp
 css/CSSVariableReferenceValue.cpp

--- a/Source/WebCore/css/CSSFontFace.cpp
+++ b/Source/WebCore/css/CSSFontFace.cpp
@@ -53,7 +53,7 @@ namespace WebCore {
 
 DEFINE_ALLOCATOR_WITH_HEAP_IDENTIFIER(CSSFontFace);
 
-template<typename T> void iterateClients(WeakHashSet<CSSFontFaceClient>& clients, T callback)
+static void iterateClients(WeakHashSet<CSSFontFaceClient>& clients, NOESCAPE const Function<void(CSSFontFaceClient&)>& callback)
 {
     for (auto& client : copyToVectorOf<Ref<CSSFontFaceClient>>(clients))
         callback(client);
@@ -249,7 +249,7 @@ static FontSelectionRange calculateItalicRange(CSSValue& value)
     ASSERT(keyword == CSSValueOblique);
     auto length = rangeValue->obliqueValues->length();
     ASSERT(length == 1 || length == 2);
-    auto angleAtIndex = [&] (size_t index) {
+    auto angleAtIndex = [rangeValue = Ref { *rangeValue }] (size_t index) {
         return Style::fontStyleAngleFromCSSValueDeprecated(*rangeValue->obliqueValues->itemWithoutBoundsCheck(index));
     };
     if (length == 1)

